packages/overture-schema-core/src/overture/schema/core/scoping/__init__.py

```diff
@@ -0,0 +1,394 @@
+...     stops=[
+...         BusStop(route='15', side=Side.LEFT, at=0.10),
+...         BusStop(route='B-Line', side=Side.RIGHT, at=0.25),
+...     ]
+... )
+
+When used on a linear feature, the side is interpreted with reference to the geometry's orientation.
+Specifically, the value `Side.LEFT` is on the left of a person who is facing forward (toward the
+end of the geometry), and `Side.RIGHT` is likewise on this person's right.
+
+
+Vehicle:
+--------
+Vehicle scoping allows a value to be tied to one or more properties of a vehicle, such as height,
+weight, or number of axles. This enables a wide variety of use cases such as restricting the
+allowed weight of trucks on bridges, limiting the allowed height of vehicles crossing under
+bridges or entering garages, applying differential speed limits to different classes of vehicle,
+*etc.* When a model is decorated with vehicle scoping, a `when.vehicle` field is automatically
+added to the model.
+
+>>> from overture.schema.core.unit import LengthUnit
+>>> from overture.schema.system.primitive import float32
+...
+>>> @scoped(Scope.VEHICLE)
+... class Fare(BaseModel):
+...     value: float32
+...
+>>> fare_schedule: list[Fare] = [
+...     Fare(
+...         value=10,
+...         when=Fare.When(vehicle=[
+...             VehicleAxleCountSelector(
+...                 dimension=VehicleDimension.AXLE_COUNT,
+...                 comparison=VehicleRelation.LESS_THAN,
+...                 value=3
+...             ),
+...             VehicleLengthSelector(
+...                dimension=VehicleDimension.LENGTH,
+...                comparison=VehicleRelation.LESS_THAN_EQUAL,
+...                value=18,
+...                unit=LengthUnit.FT,
+...             )
+...         ]),
+...     ),
+...     Fare(value=30),
+... ]
+
+
+
+Mixing scopes
+==============
+The `@scoped` decorator can mix any desired combination of scopes onto your Pydantic model.
+
+For example, suppose you are modeling a value type that can apply at certain times, along a certain
+sub-segment of a linear path, depending on whether one is traveling forward or backward along the
+path, or any combination of these three. This can be achieved easily with:
+
+>>> @scoped(Scope.TEMPORAL, Scope.GEOMETRIC_RANGE, Scope.HEADING)
+... class MyModel(BaseModel):
+...     pass
+
+
+Optional and required scopes
+============================
+When using the `@scoped` decorator, scopes are optional by default but some or all scopes may be
+made required.
+
+The following example makes all the scopes required:
+
+>>> from enum import Enum
+...
+>>> class SignalType(str, Enum):
+...    STOP_SIGN = 'stop_sign'
+...
+>>> @scoped(required=(Scope.GEOMETRIC_POSITION, Scope.HEADING))
+... class TrafficSignal(BaseModel):
+...    signal_type: SignalType
+
+The following example mixes an optional scope (temporal) with two required scopes (geometric
+position and heading).
+
+>>> @scoped(Scope.TEMPORAL, required=(Scope.GEOMETRIC_POSITION, Scope.HEADING))
```

Comment:
What are the geometric position and heading scopes relative to in this example?

Reply:
Start of the geometry.